electron_app/helpers/handleDisk.js

```diff
@@ -1,18 +1,49 @@
 const fs = require('fs');
 const mkdirp = require('mkdirp');
+const diskspace = require('diskspace');
+
+let saveConditions = { enoughSpace: Boolean, allowed: Boolean };
+let Conditions = {error: false, errorMessage: ''};
 
 const handleFolder = (directory) => {
-  return new Promise((resolve, reject) => {
-    if (!fs.existsSync(directory)) {
+    return new Promise((resolve, reject) => {
+        if (!fs.existsSync(directory)) {
+
+            mkdirp.sync(directory, (err) => {
+                if (err) {
+                    reject(err);
+                }
+            });
+        }
+        resolve(directory);
+    });
+};
 
-      mkdirp.sync(directory, (err) => {
-        if (err) {
-          reject(err);
+const handleDiskSpace = (destination, totalFileSize) => {
+    var drive = destination;
```

Comment:
I think `let` is preferred over `var` here

Reply:
Fixed!